### DISCUSSION TITLE
How to retrieve stream accuracy after experience

### DISCUSSION THREAD

**Discussion by Mattdl** [ContinualAI/avalanche#387] [Q&A] [answered]
I'm trying to save the evaluation results after an experience.
Right now I use: `results.append(strategy.eval(scenario.test_stream[:]))`

For 5-task Split-MNIST this gives me a sequence of experience (Task) accuracies, where only the last one should be considered with having the task-accuracies of all experiences:

![image](https://user-images.githubusercontent.com/13705718/110090527-b11e0380-7d97-11eb-9635-349a15efd18b.png)

However, I want to directly be able to access the stream accuracy, but in `after_eval` all the results are reset, so this isn't possible. Is there a way to still access the other evaluator plugin results, or is the only way to use a logger for this?

**Comment by vlomonaco** [comment]
Yes @Mattdl! Check out the [eval_plugin.py](https://github.com/ContinualAI/avalanche/blob/master/examples/eval_plugin.py) example and let me know if this fix your issue.
Sorry I didn't make it clearer in the doc!

**Comment by Mattdl** [comment, marked as answer]
From the example, `all_metrics = cl_strategy.evaluator.all_metrics` gives indeed access to all the metrics! Thanks!
